app/knights/equipment.py

```diff
@@ -0,0 +1,29 @@
+class Equipment:
+    """Base class for all equipment types."""
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+
+class Weapon(Equipment):
+    """Represents a weapon with a power attribute."""
+
+    def __init__(self, name: str, power: int) -> None:
+        super().__init__(name)
+        self.power = power
+
+
+class Armour(Equipment):
+    """Represents a piece of armor with a protection value."""
+
+    def __init__(self, part: int, protection: str) -> None:
+        super().__init__(part)
```

Comment:
The `Armour` class constructor is incorrectly using `part` as the parameter for the `Equipment` base class. The `super().__init__(part)` call should use a `name` parameter instead, similar to the other subclasses.